### PR TITLE
fix: FK missing for SQLite

### PR DESCRIPTION
**Pull request by billnbell**
## 🎯 Goal

fk missing for message - need SQLlite to have channel

## 🛠 Implementation details

tons of errors

### CODE REVIEW

**Review by billnbell — COMMENTED** [review]
Check array

**Comment by vishalnarkhede** [comment]
@billnbell could you give the scenario in which you managed to produce this issue/error with fk constraints failure?

In every scenario I would expect the channel to exist already when you upsert messages.

**Comment by billnbell** [comment]
> @billnbell could you give the scenario in which you managed to produce this issue/error with fk constraints failure?
> 
> In every scenario I would expect the channel to exist already when you upsert messages.

We only turn this on after a bunch of channels were already created. I open the chat, and start chatting, and the first error I get it FK constraint issue on Message. I looked into it, and it did not have the cid set / no channel.

I can think of a ton of reasons why a channel would not be in SQLLite. And then I try to chat and it says FK constraint issue. 

my fix ensures we don't get FK issue - which is what I was trying to solve for.

**Comment by billnbell** [comment]
I think another solution would be for the case when you open <Chat> you check to make sure the Channel is in SQL Lite cached.  But since I am getting it on sending a message, my fix is good.

my fix also does not add it if it is not there. So no harm right.

**Comment by vanGalilea** [comment]
Thanks @billnbell for your quick response and contribution, we will try to reproduce the issue as part of reviewing this PR and we'll keep you posted.

**Comment by billnbell** [comment]
status?

File: package/src/store/apis/upsertMessages.ts [hunk 1]
```diff
@@ -33,9 +44,15 @@ export const upsertMessages = ({
...
-  const finalQueries = [...messagesToUpsert, ...reactionsToUpsert, ...usersToUpsert];
+  const finalQueries = [
+    ...channelsToUpsert,
+    ...usersToUpsert,
+    ...reactionsToUpsert,
+    ...messagesToUpsert,
+  ];
 
   if (flush) {
+    // console.log(finalQueries, QuickSqliteClient.dbName);
```
Comment (vanGalilea) [inline package/src/store/apis/upsertMessages.ts:55]:
```suggestion
    // console.log(finalQueries, QuickSqliteClient.dbName);
```

File: package/src/store/apis/upsertMessages.ts [hunk 2]
```diff
@@ -22,6 +25,14 @@ export const upsertMessages = ({
     messagesToUpsert.push(createUpsertQuery('messages', mapMessageToStorable(message)));
     if (message.user) {
       usersToUpsert.push(createUpsertQuery('users', mapUserToStorable(message.user)));
+      if (message.cid) {
+        const channel = getChannels({ channelIds: [message.cid], currentUserId: message.user.id });
```
Comment (khushal87) [inline package/src/store/apis/upsertMessages.ts:29]:
```suggestion
        const channels = getChannels({ channelIds: [message.cid], currentUserId: message.user.id });
```

**Comment by vanGalilea** [comment]
This PR is stale, and we couldn't reproduce this issue ourselves after numerous attempts

**Pull request closed without merging**